### ISSUE TITLE
Database protection on CF stack

### ISSUE THREAD

**Issue by paolodamico** [PostHog/deployment#57]
Related to https://github.com/PostHog/posthog.com/pull/1167 & this [user report](https://posthog.slack.com/archives/C01B23A14RY/p1617411583000500) we definitely need to add some protections against accidental data deletion. Some suggestions:
- Add database protection by default to the RDS instance.
- Set a default PITR rolling backup for something low like 2-4 days. This will ensure there's something to rollback on just in case something happens and it won't add too much infrastructure expense.
- We need a way to update the CF stack without recreating the DB. If this is definitely not possible for now, can we add a warning for when you're trying to recreate your stack?

**Comment by macobo** [comment]
> We need a way to update the CF stack without recreating the DB.

This is incorrect. By default CF does not recreate the DB except when changing something that cannot be updated in place. 


Some parameters which change this are:
- PubliclyAccessible
- MasterUsername
- MasterUserPassword
- Port
- DBSubnetGroupName
- ... etc

See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-rds-database-instance.html and search for `Update requires: Replacement` for a full list.

This means that as long as we don't change the template in incompatible ways AND users don't toggle stuff inbetween deploys no recreation no replacements should occur.

Should an error occur however there is a recovery path. I previously added a `RestoreRDSSnapshotSelected` parameter to the template, which allows restoring the database from a snapshot. During replacement/deletion we always make a snapshot so that should help.

> can we add a warning for when you're trying to recreate your stack?

No. Cloudformation is extremely limited in that regard, we can't customize that. We can disable the deletion of the DB (which I _think_ should also disable replacement) is the closest we can get.

However Cloudformation does itself warn about this by showing a list of changes that apply if you wait on the page for ~30 seconds.

![image](https://user-images.githubusercontent.com/148820/113548474-44955f00-95f8-11eb-80f8-3cc8c39ff634.png)

![image](https://user-images.githubusercontent.com/148820/113548491-4bbc6d00-95f8-11eb-90fa-47de98d57974.png)

**Comment by paolodamico** [comment]
Thanks for this @macobo! Yeah so re 1, I meant for when the DB needs to be recreated, we had a user to whom this happened, and it almost caused them to lose their entire dataset. We were indeed able to recover the data through the snapshot, found the `RestoreRDSSnapshot` param when going through the stack details, didn't know you added this recently, amazing stuff!
